### PR TITLE
rmd-20676 cast boolean property

### PR DESCRIPTION
**Pull request by DavidGamba**
Gradle properties are not automatically cast to boolean in an expected
way.
See https://issues.gradle.org/browse/GRADLE-2306

### CODE REVIEW

**Comment by mch** [comment]
I think @tonytvo is best to review this, but looks ok to me.

**Review by mch — APPROVED** [review]

File: build.gradle [hunk 1]
```diff
@@ -26,7 +26,7 @@
 apply plugin: VeracodePlugin
 
 veracodeCredentials {
-    apiCredentials = "$veracodeUseAPICredentials"
+    apiCredentials = "$veracodeUseAPICredentials".toBoolean()
```
Comment (tonytvo) [inline build.gradle:29]: why do you need apiCredentials to be boolean? what is it for?

**Comment by tonytvo** [comment]
another thing to note is that we would like to commit incrementally and it's really hard to justify how much is small enough or how do we should incrementally checking it in. I'm still learning about it, here's some tips/tricks guidelines and it works for me:
1) separate formatting and feature PR (spaces, indenting, etc... in different PR with feature PR)
2) write test and put in enough functions/features to get the tests works and commit
3) bug fix (and write test for the bug if possible)
4) typically the files changed should range from 1-10 files changed for one 1 commit, however, if there are more than 20 files changed, I often ask myself, what can I do now so the next time I commit there would be a smaller number of files changed (there should only be 1 place to make the change, instead of changing too many places)

**Comment by DavidGamba** [comment]
Thanks Tony, this is a fix for the previous PR I sent, which included the API functionality.

I am using apiCredentials as a boolean property to know if the user wants to use API Credentials or username and password credentials. The previous PR introduced the functionality but I missed forcing this property to a boolean then.

**Comment by DavidGamba** [comment]
On the tests side of things. Testing to ensure a passed parameter is of the expected type is kind of excessive but to some degree required in annoying dynamic languages like groovy that accept a string as a boolean (I am not going to add a test for that though, where does it end!).

I am still trying to figure out how to do tests for this plugin since all it does is to call a third party library for all its tasks.

**Comment by tonytvo** [comment]
I see ...
1) regarding to apiCredentials as a boolean, I would ask do we need this apiCredentials to tell whether the user want APICredentials or userNamePasswordCredentials
 - we could configure UserNamePasswordCredentials by default. If it isn't configured then we look to see if APICredentials is configured and set the flag "isAPICredentialsUsed" = true and use that flag for later purposes.

2) so the basic idea is when you run a task, you want to know that if a specific action has been invoked (call 3rd party APIs), you want to make sure that the right method has been called with the right arguments. This is where you test the interaction between the code with the 3rd party API, you want to make sure it behaves as you expect. (groovy spock framework make this easy to work with for developers, you can take a look at the tests in resmd-packaging/buildSrc/groovy/com.....)
  - In most of the times, you would like to test if the system behaves as expected by inspecting the state of the system after the method execution. However, this is not possible when 3rd party system is not available  (like veracode system, like methods call go over the network, databases, etc...), as a result. You would assume that API of the 3rd party is correct and we test the interaction between the system with the 3rd party API. This is where most of the tests of veracode gradle plugins could fit in.
  - note that the reason most people don't like this interaction style is you could "over specify" the software with the interaction tests and when the API changed, lots of the tests will break.

**Pull request closed without merging**